Migrate rxjava to rxjava2

joshuadeguzman opened the pull request:
**Summary**
- Migrated the SDK from RxJava1 to RxJava2
- Improved Unit Tests
- Minor clean ups

**Related issues**
#21

joshuadeguzman commented on .gitignore:
```diff
@@ -1,6 +1,6 @@
 *.iml
 .gradle
-/local.properties
+local.properties
```
Yes, the current setup of gitignore listen to the changes of `local.properties`. I have update the path referencing to it.

joshuadeguzman commented:
> `Update respose interceptor` typo in commit and I think it can be squashed to the previous commit since it has the same file changes.

Great, will update this.

joshuadeguzman commented on local.properties:
```diff
@@ -1,10 +1,8 @@
-# Do not modify this file -- YOUR CHANGES WILL BE ERASED!
-#
-# This file must *NOT* be checked into Version Control Systems,
+## This file must *NOT* be checked into Version Control Systems,
```
Will revert this, this should have been ignored. This happens because the current codebase is not ignoring changes made in `local.properties` file.

gumil commented:
> > `Update respose interceptor` typo in commit and I think it can be squashed to the previous commit since it has the same file changes.
> 
> Great, will update this.

Nice! Also, who's maintaining this? I also see another PR not yet merged.

joshuadeguzman commented:
> > > `Update respose interceptor` typo in commit and I think it can be squashed to the previous commit since it has the same file changes.
> > 
> > 
> > Great, will update this.
> 
> Nice! Also, who's maintaining this? I also see another PR not yet merged.

Hi @gumil, unfortunately, I don't really know as of the moment, @cwliu perhaps?. I just went through this repo after work. Heh.

joshuadeguzman commented:
Hi @gumil,

I have updated my diff. Had to force push to rewrite the typo in the commit messages and squashed retrofit migration (13a8861). I take into consideration the careful force push usage in this PR by using `git push --force-with-lease`.

Let me know if you have more suggestions, thanks!
